src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,115 @@
+//Author: Jonathan Jarman
+//Date: 6/11/2023
+//File: String Manipulation
+/*The StringManipulation class provides a set of methods for manipulating strings,
+such as counting the number of words, removing characters at specific positions,
+extracting substrings, and restoring shuffled strings. It implements the StringManipulationInterface,
+which defines the contract for these operations.
+* */
 public class StringManipulation implements StringManipulationInterface {
 
+    String string; // holding string input
+
     @Override
-    public String getString() {
-        return null;
+    public String getString() { // method to return string
+        return string;
     }
 
     @Override
-    public void setString(String string) {
+    public void setString(String string) { // method to set the string
+        this.string = string;
     }
 
+    // this method will count the # of words in the string
     @Override
     public int count() {
-        return 0;
+        // throw null pointer exception if string is null
+        if (string == null) {
+            throw new NullPointerException("String is null");
+        }
+        // if string is empty throw an exception
+        if (string.isEmpty()) {
+            throw new IndexOutOfBoundsException("string is empty!");
```

Comment:
if string is empty, count should be 0 not an exception

Reply:
edited method and test case.